src/lib/utils/settings/abstract_setting.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+namespace opossum {
+
+/**
+ * This is an abstract class for all settings objects.
+ *
+ * Settings register itself at the SettingsManager.
+ * Therefore it is neccessary that they are owned by a shared pointer (instantiate with std::make_shared).
+ * Thus, you need to call unenroll() when the setting is not longer needed (it won't get destructed otherwise,
+ * as SettingsManager would still hold a shared pointer to it).
+ *
+ * Settings provide a getter/setter with a string.
+ * When the set(...) method ia called, a setting can invoke a method of its parent that triggers
+ * the immediate appliance of the changed value.
+ *
+ * Settings have a unique name that consists of the name of its parent and the setting
+ * name (e.g. "TaskScheduler.workers")
```

Comment:
The concept of a parent should probably be introduced above.

Reply:
I have "A setting is an adjusting screw of any component with a variable value. That component (or "parent") instantiates and destroys a setting." now.

Reply:
Sounds good. Often, the term "knob" is used for database systems.